LanguageExt.Core/TypeClasses/MonadResource/MonadResource.cs

```diff
@@ -0,0 +1,13 @@
+using LanguageExt.Attributes;
+using System;
+
+namespace LanguageExt.TypeClasses
+{
+    [Typeclass]
+    public interface MonadResource : Typeclass
```

Comment:
I notice `Use` and `Release` both have a generic of `R` - shouldn't the interface be `MonadResoure<R>` - is there a benefit to applying it on the methods?

Reply:
I think I see now.  `MonadResource<R>` would force anything implementing it to have its generic parameter constrained to `IDisposable`, and so this allows _bolting on_ of `Use` and `Release` to any monadic type.

Reply:
I actually didn't really consider applying `MonadResource` to other types, but I like the idea and I think that's what you're working on below.